b2g/components/RemoteControlService.jsm

```diff
@@ -950,6 +990,13 @@ EventHandler.prototype = {
     });
   },
 
+  // Synthesize PIN for J-PAKE round 2 by original PIN
+  // and TLS server cert fingerprint
+  _synthesizeJPAKEPIN: function(aPIN) {
+    return SHA256(aPIN + this._server.getCertFingerprint())
+           .slice(-1 * JPAKE_PIN_LENGTH);
```

Comment:
You slice -32 here right? Not sure if that's what you want. The result of SHA256 should be 32 bytes. So if it's bytes you're not doing anything and if it were bits, you'd get only 32 bit. I'm a bit confused...